src/Facebook/Url/FacebookUrlDetectionHandler.php

```diff
@@ -160,4 +160,23 @@ protected function getHeader($key)
     {
         return $this->getServerVar('HTTP_' . $key);
     }
+
+    /**
+     * Checks if the value in X_FORWARDED_HOST is a valid hostname
+     * Could prevent unintended redirections
+     *
+     * @param string $header
+     *
+     * @return boolean
+     */
+    protected function isValidForwardedHost($header)
+    {
+        $elements = explode(',', $header);
+        $host = $elements[count($elements) - 1];
+        
+        return preg_match("/^([a-z\d](-*[a-z\d])*)(\.([a-z\d](-*[a-z\d])*))*$/i", $host) //valid chars check
```

Comment:
@tgallice No it can't, your regex will allow a host to start with `.` or `-`.
BTW I think it can be replace with something like `/^[a-z\d][a-z\d-]*(?:\.[a-z\d][a-z\d-]*)*$/i`, what do you think about it? ping @gfosco @SammyK